Java/core/src/main/java/com/amazon/randomcutforest/state/tree/CompactRandomCutTreeState.java

```diff
@@ -25,7 +25,7 @@
 public class CompactRandomCutTreeState {
     public LeafStoreState leafStoreState;
     public NodeStoreState nodeStoreState;
-    public short rootIndex;
+    public int rootIndex;
```

Comment:
Yes. It probably makes sense that the trees assume int/double for all variables (so that we do not copy codes for int/short). The specific storage classes can perform the conversations (if needed)